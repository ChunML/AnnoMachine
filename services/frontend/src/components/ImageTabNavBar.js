import React from 'react';
import PropTypes from 'prop-types';

function ImageTabNavbar({ selectedTab, onTabChange }) {
  return (
    <div
      className="image-nav flex-center space-around"
      style={{ width: '100%' }}
    >
      <div
        className={`${selectedTab === 'all' ? 'active' : ''} image-nav-link`}
        onClick={() => onTabChange('all')}
      >
        All Images
      </div>
      <div
        className={`${selectedTab === 'yours' ? 'active' : ''} image-nav-link`}
        onClick={() => onTabChange('yours')}
      >
        Yours
      </div>
    </div>
  );
}

ImageTabNavbar.propTypes = {
  selectedTab: PropTypes.string.isRequired,
  onTabChange: PropTypes.func.isRequired,
};

export default ImageTabNavbar;
