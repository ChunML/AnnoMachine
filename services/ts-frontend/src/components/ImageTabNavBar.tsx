import React from 'react';
import PropTypes from 'prop-types';

interface ImageTabNavbarProps {
  selectedTab: string;
  onTabChange(tabName: string): void;
}

function ImageTabNavbar({ selectedTab, onTabChange }: ImageTabNavbarProps) {
  return (
    <div
      className="image-nav flex-center space-around"
      style={{ width: '100%' }}
    >
      <div
        className={`${selectedTab === 'all' ? 'active' : ''} image-nav-link`}
        onClick={() => onTabChange('all')}
      >
        All Images
      </div>
      <div
        className={`${selectedTab === 'yours' ? 'active' : ''} image-nav-link`}
        onClick={() => onTabChange('yours')}
      >
        Yours
      </div>
    </div>
  );
}

ImageTabNavbar.propTypes = {
  selectedTab: PropTypes.string.isRequired,
  onTabChange: PropTypes.func.isRequired,
};

export default ImageTabNavbar;
