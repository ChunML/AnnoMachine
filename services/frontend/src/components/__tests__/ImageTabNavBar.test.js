import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ImageTabNavBar from '../ImageTabNavBar';
import '../../setupTests';

describe('Tab all images selected', () => {
  const component = (
    <ImageTabNavBar selectedTab="all" onTabChange={jest.fn()} />
  );

  it('ImageTabNavBar renders properly', () => {
    const wrapper = shallow(component);
    const menu = wrapper.find('.image-nav');
    expect(menu.length).toBe(1);
    const tabs = menu.find('.image-nav-link');
    expect(tabs.length).toBe(2);
    expect(tabs.get(0).props.className).toEqual('active image-nav-link');
    expect(tabs.get(1).props.className).toEqual(' image-nav-link');
    expect(component.props.onTabChange).toHaveBeenCalledTimes(0);
    tabs.at(0).simulate('click');
    expect(component.props.onTabChange).toHaveBeenCalledTimes(1);
    tabs.at(1).simulate('click');
    expect(component.props.onTabChange).toHaveBeenCalledTimes(2);
  });

  it('ImageTabNavBar renders a snapshot properly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Tab your images selected', () => {
  const component = (
    <ImageTabNavBar selectedTab="yours" onTabChange={jest.fn()} />
  );

  it('ImageTabNavBar renders properly', () => {
    const wrapper = shallow(component);
    const menu = wrapper.find('.image-nav');
    expect(menu.length).toBe(1);
    const tabs = menu.find('.image-nav-link');
    expect(tabs.length).toBe(2);
    expect(tabs.get(0).props.className).toEqual(' image-nav-link');
    expect(tabs.get(1).props.className).toEqual('active image-nav-link');
    expect(component.props.onTabChange).toHaveBeenCalledTimes(0);
    tabs.at(0).simulate('click');
    expect(component.props.onTabChange).toHaveBeenCalledTimes(1);
    tabs.at(1).simulate('click');
    expect(component.props.onTabChange).toHaveBeenCalledTimes(2);
  });

  it('ImageTabNavBar renders a snapshot properly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
