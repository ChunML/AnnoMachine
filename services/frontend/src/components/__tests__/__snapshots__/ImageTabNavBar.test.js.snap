// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab all images selected ImageTabNavBar renders a snapshot properly 1`] = `
<div
  className="ui tabular menu"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <a
    className="active item"
    onClick={[Function]}
  >
    All Images
  </a>
  <a
    className=" item"
    onClick={[Function]}
  >
    Yours
  </a>
</div>
`;

exports[`Tab your images selected ImageTabNavBar renders a snapshot properly 1`] = `
<div
  className="ui tabular menu"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <a
    className=" item"
    onClick={[Function]}
  >
    All Images
  </a>
  <a
    className="active item"
    onClick={[Function]}
  >
    Yours
  </a>
</div>
`;
